docs/reference/mapping/types/flattened.asciidoc

```diff
@@ -407,8 +423,16 @@ For example:
 ----
 PUT idx
 {
+  "settings": {
+    "index": {
+      "mapping": {
+        "source": {
+          "mode": "synthetic"
+        }
+      }
+    }
+  },
   "mappings": {
-    "_source": { "mode": "synthetic" },
```

Comment:
I don't fully understand why this failures happens for flattened field doc tests: https://buildkite.com/elastic/elasticsearch-pull-request/builds/36289#01928be4-ffbb-4def-8006-256e92e1e5d5/6-13640

`foo` is returned in an extra array. Strange that this failure happens with a doc change.